Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ButtonGroup.cpp

```diff
@@ -31,11 +31,22 @@ namespace AzToolsFramework::ViewportUi::Internal
     {
         if (auto buttonEntry = m_buttons.find(buttonId); buttonEntry != m_buttons.end())
         {
-            for (auto& button : m_buttons)
-            {
-                button.second->m_state = Button::State::Deselected;
-            }
+            ClearHighlightedButton();
             buttonEntry->second->m_state = Button::State::Selected;
+            m_highlightedButtonid = buttonId;
+        }
+    }
+
+    void ButtonGroup::ClearHighlightedButton()
+    {
+        if (m_highlightedButtonid == InvalidButtonId)
+        {
+            return;
+        }
+        if (auto buttonEntry = m_buttons.find(m_highlightedButtonid); buttonEntry != m_buttons.end())
+        {
+            buttonEntry->second->m_state = Button::State::Deselected;
+            m_highlightedButtonid = InvalidButtonId;
         }
```

Comment:
Nice! This is looks cool - I think `m_highlightedButtonid` should be `m_highlightedButtonId` (lowercase `i`) though

Reply:
oops!